packages/model/src/tests/workspace.test.ts

```diff
@@ -0,0 +1,82 @@
+import {
+  Workspace,
+  WorkspaceOptions,
+  ProjectKind,
+  Service,
+  WebApp,
+  Package,
+} from '..';
+import {Base, Options} from '../base';
+
+// Test creation of different types of workspaces
+describe('Workspace', () => {
+  const name = 'test';
+  const root = '../../';
+
+  const options: Options = {
+    name,
+    root,
+  };
+  const base = new Base(options);
+  const webApp = new WebApp(options);
+  const pkg = new Package(options);
+  const service = new Service(options);
+  const webApps: WebApp[] = [webApp];
+  const packages: Package[] = [pkg];
+  const services: Service[] = [service];
+  const emptyWebApps: WebApp[] = [];
+  const emptyPackages: Package[] = [];
+  const emptyServices: Service[] = [];
+
+  it('creates a workspace for a WebApp', () => {
+    const opts: WorkspaceOptions = {
+      webApps,
+      services: emptyServices,
+      packages: emptyPackages,
+      name,
+      root,
+    };
+    const webAppWorkspace = new Workspace(opts);
+    expect(webAppWorkspace.projects[0].kind).toBe(ProjectKind.WebApp);
+    expect(webAppWorkspace.private).toBe(true);
+  });
+  it('creates a workspace for a Package', () => {
+    const opts: WorkspaceOptions = {
+      webApps: emptyWebApps,
+      services: emptyServices,
+      packages,
+      name,
+      root,
+    };
+    const packageWorkspace = new Workspace(opts);
+    expect(packageWorkspace.projects[0].kind).toBe(ProjectKind.Package);
+    expect(packageWorkspace.private).toBe(false);
```

Comment:
Not entirely sure what `private` is intended for given how vague it's named. Perhaps @lemonmade can shine some light.

Reply:
What I inferred from the code is that a workspace with only a package or packages would be public, i.e packages that could be published, every other kind of workspace would be private.  Not sure if that was what was intended?